gossip/c_block_callbacks.go

```diff
@@ -183,7 +183,7 @@ func consensusCallbackBeginBlockFn(
 
 				// Execute pre-internal transactions
 				preInternalTxs := blockProc.PreTxTransactor.PopInternalTxs(blockCtx, bs, es, sealing, statedb)
-				preInternalReceipts := evmProcessor.Execute(preInternalTxs)
+				preInternalReceipts := evmProcessor.Execute(preInternalTxs, common.Hash{})
```

Comment:
Instead of adding the prev-randao to the `Execute` function I think it would be a better fit to pass it to the `Start` function (for instance in line 181) instead, since all the other context information is also set there.

Reply:
That was my initial thought, but the necessary events are not yet ready. I might be missing something, but I dont think passing `prevrandao` to `Start` is possible...

Reply:
I believe lines 226-239 need to be executed before Im able to call `GetPrevRandao()` https://github.com/Fantom-foundation/Sonic/blob/8fb94a51d66590e10fac2c2100a4dbd784d0fa48/gossip/c_block_callbacks.go#L226

Reply:
Nevertheless, I would like to make a Norma run to doublecheck that the block processing works correctly